docs/explanations/how_to_interpret_output_of_fix_plan_api.md

```diff
@@ -0,0 +1,222 @@
+  The JSON representation for this step is the following one:
+
+  ```json
+  {
+    "data": {
+      "reason": "required-pro-service"
+    },
+    "operation": "attach",
+    "order": 1
+  }
+  ```
+
+  The reason field details why an attach is needed. This can be either because the user has
+  an expired subscription or the fix requires a specific Pro service to upgrade an affected
+  package.
+
+* **Enable**: This operation details that the user needs to enable a specific Pro service.
+  The JSON representation for this step is the following one:
+
+  ```json
+  {
+    "data": {
+      "service": "esm-infra"
+    },
+    "operation": "enable",
+    "order": 1
+  }
+  ```
+
+  The data object contains which service will need to be enabled.
+
+* **NoOp**: This indicates that no operation need to be performed to address the CVE.
+  The JSON representation for this step is the following one:
+
+  ```json
+  {
+    "data": {
+      "status": "system-not-affected"
+    },
+    "operation": "no-op",
+    "order": 1
+  }
+  ```
+  
+  The data object will state why no operation is needed. This can either be because the CVE doesn't
+  affect the system or because the CVE is already fixed in the machine.
+
+## What type of warnings can be generated ?
+
+There are two distinct warning that can happen when executing the plan API:
+
+* **Package cannot be installed**: This happens when the endpoint identifies that a given package
+  cannot be installed (i.e. the user lacks a required APT source in the machine). This can be seen
+  in the following JSON representation:
+
+  ```json
+  {
+    "data": {
+      "binary_package": "binary_package",
+      "binary_package_version": "3.5.12-1ubuntu7.16",
+      "source_package": "source_package"
+    },
+    "order": 1,
+    "warning_type": "package-cannot-be-installed"
+  }
+  ```
+
+ The data object details the packages that cannot be installed and the version of the package.
+
+* **Security issue not fixed**: This happens when the CVE doesn't provide a fix for some of the
+  affected packages. This can be seen in the following JSON representation:
+
+  ```json
+  {
+    "data": {
+      "source_packages": ["source_package"],
+      "status": "needs-triage"
+    },
+    "order": 1,
+    "warning_type": "security-issue-not-fixed"
+  }
+  ```
+
+
+## What errors can be generated ?
+
+There are two errors that can occur when running this API endpoint. Those errors happen when a
+CVE has an invalid format or if it doesn't exist. When an error happen, it will
+be described by the following JSON representation:
+
+```json
+{
+  "code": "security-fix-not-found-issue", 
+  "msg": "Error: CVE-XXXX-XXXXX not found."
+}
+```
+
+
+## Why do we need the order attribute ?
+
+The order attribute is used to allow the user to properly understand the exact sequence of events
+that will happen if the fix would actually be performed. This is specially true for situations where
+we have warnings, that sit outside the plan object.
```

Comment:
I'm not sure what you mean by "sit outside the plan object"?

Reply:
It is just that warning live in another attribute than the plan attribute. But let me know If I should be clearer here